BraintreeCore/src/main/java/com/braintreepayments/api/UserCanceledException.java

```diff
@@ -5,7 +5,28 @@
  */
 public class UserCanceledException extends BraintreeException {
 
+    private boolean isExplicitCancellation;
+
     UserCanceledException(String message) {
         super(message);
     }
+
+    UserCanceledException(String message, boolean isExplicitCancellation) {
+        super(message);
+        this.isExplicitCancellation = isExplicitCancellation;
+    }
+
+    /**
+     * @return whether or not the user explicitly canceled the payment flow.
+     *
+     * {@link true} if the user manually confirms cancellation of the payment flow.
+     *
+     * {@link false} if the user returns to the app without completing the payment flow and the
+     * action performed to return to the app is unknown. For browser switching flows, this could
+     * mean the user returned to the app through multi-tasking without completing the flow, the user
+     * closed the browser tab, or the user pressed the back button.
+     */
+    public boolean isExplicitCancellation() {
```

Comment:
For browser switching flows, we can't distinguish between a user returning to the app through multi tasking and not completing the payment flow, or actually closing the browser (x or back button). We tried to express that this boolean will be true only if we _know_ that the user explicitly cancelled. 

Open to suggestions on how to make this as clear as possible

Reply:
I think that's the crux of the issue. Chrome Custom Tabs doesn't make that information available. We only know that the user has returned to the screen.

Reply:
In my opinion we shouldn't have a canceled event for when the user returns back to the screen. It's a bit premature to say the User has "canceled" the flow. They could be multi-tasking while completing a purchase.